typing/env.ml

```diff
@@ -808,6 +821,10 @@ let print_structure_components_reason ppf = function
   | Project -> Format.fprintf ppf "have any components"
   | Open -> Format.fprintf ppf "be opend"
 
+type none_in_quotations_context =
```

Comment:
I think we can extend this enum to include all items that are affected by stages, and attach this information to `Incompatible_stage`.

Could you leave a CR-someday for future work?